core-ui/src/components/Extensibility/migration.js

```diff
@@ -0,0 +1,60 @@
+import * as jp from 'jsonpath';
+import { cloneDeep } from 'lodash';
+
+const SUPPORTED_VERSIONS = [
+  { version: '0.4', latest: false },
+  { version: '0.5', latest: true },
+];
+
+export const getSupportedVersions = () =>
+  SUPPORTED_VERSIONS.map(version => version.version).sort();
+
+export const getLatestVersion = () =>
+  SUPPORTED_VERSIONS.find(version => version.latest)?.version;
+export const formatCurrentVersion = version => {
+  if (!version) return null;
+  return (
+    version
+      .toString()
+      .replaceAll("'", '')
+      .replaceAll('"', '') || ''
+  );
+};
+
+export function migrateToLatest(resource) {
+  if (!resource) return undefined;
+
+  const newestVersion = getLatestVersion();
+  const currentVersion = formatCurrentVersion(resource?.data?.version);
+
+  const functionName = currentVersion.replace('.', '');
```

Comment:
```suggestion
  const functionName = currentVersion?.replace('.', '');
```
we should have a check if the current version even exists